### PR TITLE
Fix issues with DRK multi-hit and resource modules

### PR DESCRIPTION
**Pull request by AzariahAeroborn**

### CODE REVIEW

**Comment by AzariahAeroborn** [comment]
I intentionally have not updated the changelog yet.  I'm working on one more change that will change "no resource gain on miss/invuln" handling to use the hit type enum rather than checking for damage = 0, but wanted to split that into a separate pull request because it will impact core.  I will update the changelog to include all three changes with that pull.

File: src/parser/jobs/drk/modules/ResourceSimulator.js [hunk 1]
```diff
@@ -179,6 +179,12 @@ export default class Resources extends Module {
 	}
 
 	checkBloodOvercap(actionBloodChange) {
+		if (actionBloodChange < 0) {
+			// Spender
+			if (this.combatants.selected.hasStatus(STATUSES.DELIRIUM.id)) {
+				actionBloodChange = 0
```
Comment (ackwell) [inline src/parser/jobs/drk/modules/ResourceSimulator.js]: _Generally_ avoid modifying function args, pretend they're `const`. In this particular case, the `if` below checks against this value, so you can probably afford to

```suggestion
				return
```

to signify there's no consumption

File: src/parser/jobs/drk/modules/ResourceSimulator.js [hunk 2]
```diff
@@ -222,27 +228,34 @@ export default class Resources extends Module {
...
 			const actionInfo = RESOURCE_GENERATORS[abilityId]
-			if (!actionInfo.requiresCombo || event.type === 'combo') {
+			if ((!actionInfo.requiresCombo && event.type !== 'combo') || event.type === 'combo') {
 			// Only gain resources if the action does not require a combo or is in a valid combo
 				actionBloodGain += RESOURCE_GENERATORS[abilityId].blood
 				actionMPGain += RESOURCE_GENERATORS[abilityId].mp
 			}
 		}
 
+		console.log(`Net Blood Change: ${actionBloodGain}`)
```
Comment (ackwell) [inline src/parser/jobs/drk/modules/ResourceSimulator.js]: logging~

Comment (AzariahAeroborn) [reply src/parser/jobs/drk/modules/ResourceSimulator.js]: Whoops.  Removed.

File: src/parser/jobs/drk/modules/ResourceSimulator.js (hunk 1 above)
Comment (AzariahAeroborn) [reply src/parser/jobs/drk/modules/ResourceSimulator.js]: Change updated, also consolidated the double-IF into a single and statement.

**Pull request merged**